src/tools/illink/src/linker/Linker.Dataflow/SystemTypeValue.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TypeSystemProxy;
+using Mono.Linker;
+
+namespace ILLink.Shared.TrimAnalysis
+{
+	internal sealed partial record SystemTypeValue : SingleValue
+	{
+		public SystemTypeValue (in TypeProxy representedType, ITryResolveMetadata resolver)
+		{
+			RepresentedType = representedType;
+			this.resolver = resolver;
+		}
+
+		private readonly ITryResolveMetadata resolver;
+
+		public bool Equals (SystemTypeValue? other) => other is not null && TypeReferenceEqualityComparer.AreEqual (RepresentedType.Type, other.RepresentedType.Type, resolver);
```

Comment:
Now that `TypeProxy` uses a `TypeReference`, should we instead push the `Equals` override (and resolver field) into `TypeProxy`? Then we might be able to leave `SystemTypeValue` in one shared file.